controllers/nginx/pkg/cmd/controller/main.go

```diff
@@ -31,29 +32,32 @@ func main() {
 	ngx := newNGINXController()
 	// create a custom Ingress controller using NGINX as backend
 	ic := controller.NewIngressController(ngx)
-	go handleSigterm(ic)
+
 	// start the controller
-	ic.Start()
-	// wait
-	glog.Infof("shutting down Ingress controller...")
-	for {
-		glog.Infof("Handled quit, awaiting pod deletion")
-		time.Sleep(30 * time.Second)
-	}
-}
+	go ic.Start()
 
-func handleSigterm(ic *controller.GenericController) {
 	signalChan := make(chan os.Signal, 1)
 	signal.Notify(signalChan, syscall.SIGTERM)
 	<-signalChan
+
 	glog.Infof("Received SIGTERM, shutting down")
 
-	exitCode := 0
 	if err := ic.Stop(); err != nil {
-		glog.Infof("Error during shutdown %v", err)
-		exitCode = 1
+		glog.Errorf("unexpected error shutting down the ingress controller: %v", err)
 	}
 
-	glog.Infof("Exiting with %v", exitCode)
-	os.Exit(exitCode)
+	glog.Infof("stopping nginx gracefully...")
+	ngx.Stop()
+
+	timer := time.NewTicker(time.Second * 1)
+	for {
+		select {
+		case <-timer.C:
+			if !isNginxRunning(ngx.ports.Status) {
```

Comment:
@aledbf This is not working as expected, as Nginx won't accept new connection while shutting down, therefore giving the impression that it has stopped.